### PR TITLE
Feat: implement product delete.

### PR DESCRIPTION
**Pull request by chirino**

### CODE REVIEW

**Comment by chirino** [comment]
This is a task of https://github.com/trustification/trustify/issues/563

**Comment by chirino** [comment]
mutation added.

File: client/src/app/queries/products.ts [hunk 1]
```diff
@@ -41,3 +47,21 @@ export const useFetchProductById = (id?: number | string) => {
...
 };
+
+export const useDeleteProductByIdMutation = () => {
+  const queryClient = useQueryClient();
+  const {pushNotification} = React.useContext(NotificationsContext);
+
+  return useMutation({
+    mutationFn: (id: number | string) => deleteProductById(id),
+    onSuccess: (_res, id) => {
+      queryClient.invalidateQueries({queryKey: [ProductsQueryKey, id]});
```
Comment (gildub) [inline client/src/app/queries/products.ts:58]: The invalidateQueries key could actually be :
`queryKey: [SBOMsQueryKey]`

**Review by gildub — CHANGES_REQUESTED** [review]
LGTM just a minor change, please see inline.

**Review by carlosthe19916 — CHANGES_REQUESTED** [review]
Thanks for the PR. The same comments done at https://github.com/trustification/trustify-ui/pull/120#pullrequestreview-2204769060 apply here

**Comment by chirino** [comment]
rolled into https://github.com/trustification/trustify-ui/pull/129

**Pull request closed without merging**